x/superfluid/keeper/unpool.go

```diff
@@ -99,39 +102,48 @@ func (k Keeper) checkUnpoolWhitelisted(ctx sdk.Context, poolId uint64) error {
 	return types.ErrPoolNotWhitelisted
 }
 
-// check if pool is whitelisted for unpool
-func (k Keeper) validateLockForUnpool(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, lockId uint64) (*lockuptypes.PeriodLock, error) {
+// validateGammLockForSuperfluid checks if the provided lock:
+// 1) is owned by the provided sender
+// 2) contains only 1 coin
+// 3) contains the gamm LP shares associated with the provided poolId
+func (k Keeper) validateGammLockForSuperfluid(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, lockId uint64) (*lockuptypes.PeriodLock, error) {
```

Comment:
by superfluid, are you referring to validating the lock and making sure it can be superfluid staked? If so, consider changing to `validateGammLockForSuperfluidStaking`

Reply:
I am a little worried that the new name might be misleading since that just because a lock has met the conditions below, it does not mean that it is safe to be superfluid staked 🤔

Reply:
Other than being whitelisted, what other checks need to happen to make a lock safe for superfluid staking? It feels like if the above checks are valid the lock is safe for superfluid staking (given it is whitelisted)

Reply:
Renamed it to `validateGammLockForSuperfluidStaking` here https://github.com/osmosis-labs/osmosis/pull/5178/commits/f8756cd1bd107867fb03005adf907ad55c2f9573